File: src/ed/ogonzalezm/a06/CircularList.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ed.ogonzalezm.a06;

/**
 *
 * @author HONORIO ZAIBACK
 */
public class CircularList <T> {
    Nodo<T> pivote;
    Nodo<T> pointer;
    int length;
    
    CircularList(){
        pivote = pointer = null;
        length=0;
    }
    
    CircularList(T d){
        Nodo<T> node = new  Nodo<T>(d);
        pivote = node.next = node;
        length++;
    }
    
    boolean isEmpty(){
        return length==0;
    }
    
    void insert(T d){
        Nodo<T> node = new  Nodo<T>(d);
        if(isEmpty()){
            pivote = node.next = node;
        }else{
            node.next = pivote.next;
            pivote.next = node;
        }
        length++;
    }
    
    void deleteNodo(T d){
        if(!isEmpty()){
            pointer = pivote;
            for(int i=1; i<=length;i++){
                if(pointer.data == d){
                    pointer.next =pointer.next.next;
                }
                length--;
                pointer = null;
                i = length + 1;
            }
            pointer = pointer.next;
        }
    }
    
    void deleteList(){
        pivote = pointer = null;
    }
}

File: src/ed/ogonzalezm/a06/CircularList.cpp
#include "CircularList.h"

template <typename T>
CircularList<T>::CircularList() {
	pivote = pointer = NULL;
	length = 0;
}

template <typename T>
CircularList<T>::CircularList(T d) {
	nodo<T> *node = new nodo<T>(d);
	pivote = node->next = node;
	length++;
}

template <typename T>
CircularList<T>::~CircularList(){}

template <typename T>
bool CircularList<T>::isEmpty() {
	return length == 0;
}

template <typename T> 
void CircularList<T>::insert(T d) {
	nodo<T> node* = new nodo<T>(d);
	if (isEmpty()) {
		pivote = node->next = node;
	}else{ 
		node->next = pivote->next;
		pivote->next = node;
	}
	length++;
}

template <typename T>
void CircularList<T>::deleteNodo(T d){
	if (!isEmpty()) {
		pointer = pivote;
		for (int i = 1; i <= length; i++) {
			if (pointer->data == d) {
				pointer -> next = pointer->next->next;
				if (pivote->data == d) {
					pivote = pointer;
				}
				length--;
				pointer = NULL;
				i = length + 1;
			}
			pointer = pointer->next;
		}
	}
}

template <typename T>
void CircularList<T>::deleteList(){
	pivote = pointer = NULL;
}

template <typename T>
void CircularList<T>::showList() {
	std::cout()
}
